### PR TITLE
FEATURE: Add support for CORS

### PR DESCRIPTION
**Pull request by dfeyer**

### CODE REVIEW

File: Classes/Http/HttpCorsComponent.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Http\Component\ComponentChain;
+use Neos\Flow\Http\Component\ComponentContext;
+use Neos\Flow\Http\Component\ComponentInterface;
+
+/**
+ * A simple HTTP Component to configure CORS for the GraphQL endpoint
+ */
+class HttpCorsComponent implements ComponentInterface
```
Comment (dfeyer) [inline Classes/Http/HttpCorsComponent.php:12]: Can be renamed to something more generic, because it can add any header

**Comment by bwaidelich** [comment]
@dfeyer Thanks for this addition - but why not add https://github.com/Flowpack/Flowpack.Cors ?
Or - to put it differently - if using `Flowpack.Cors` (which I often do) the headers would be added twice

**Comment by bwaidelich** [comment]
@dfeyer *ping* :)

**Comment by dfeyer** [comment]
@bwaidelich Basically the Flowpack.Cors add the header globally and it's not what I need on many project, I want to have granular configuration to enable the feature per URI.

> if using `Flowpack.Cors` (which I often do) the headers would be added twice

Yes, that's right, so maybe both package, should check if the headers are set or not, and skip if an other package set the header ... but it's hard to follow for developpers.

**Comment by bwaidelich** [comment]
> the Flowpack.Cors add the header globally

True, but then we should extend that package so that it can work with filters (we might even re-use the Flow Request Patterns

> it's hard to follow for developpers

We could add it to the suggested packages in the composer manifest and add a paragraph to the README

**Comment by dfeyer** [comment]
I agree, will try to push a PR on Flowpack.Cors next week

**Comment by bwaidelich** [comment]
Thanks for the update, I'm looking forward to your change on Flowpack.Cors
